Cython/Compiler/PyrexTypes.py

```diff
@@ -4147,8 +4148,12 @@ class CEnumType(CIntLike, CType):
     signed = 1
     rank = -1  # Ranks below any integer type
 
-    def __init__(self, name, cname, typedef_flag, namespace=None):
+    def __init__(self, name, cname, typedef_flag, namespace=None, doc=None):
         self.name = name
+        if doc is None:
+            self.doc = "An enumeration."
+        else:
+            self.doc = doc
```

Comment:
The string constant would need to be an `StringEncoding.EncodedString()`, but rather, I think we should inherit the docstring from the superclass here and not set a default.
```suggestion
        self.doc = doc
```
I understand that this makes the test a little more difficult, but I'd rather not make a test rely on the docstring of Python's `IntEnum` class at all, so I'm fine with a test that the docstrings set for other enum classes do not leak into the enums that have no docstring, and not testing for a specific string value.

Reply:
I'm not sure I fully understood the second half of the paragraph. But I hadn't realized that the `"An enumeration."` originates from python's Enum - I couldn't figure out where it came from that's why I originally kept it. So yeah, it makes sense to me to keep the default `__doc__` to `None`.

Reply:
Hmm I just noticed that my changes actually set the docstring to `None`, don't they? That wasn't intended.